libdnf/transaction/db/db.cpp

```diff
@@ -77,8 +77,16 @@ std::unique_ptr<libdnf::utils::SQLite3> transaction_db_connect(libdnf::Base & ba
     std::string db_path;
 
     // use db_path based on persistdir
-    auto persistdir = base.get_config().persistdir().get_value();
-    std::filesystem::path path(persistdir);
+    auto & config = base.get_config();
+    // TODO(jrohel): Lock "installroot" after read its value
+    auto installroot = config.installroot().get_value();
+    std::filesystem::path persistdir = config.persistdir().get_value();
+    std::filesystem::path path;
+    if (installroot.size() > 1) {
+        path = installroot;
+    }
```

Comment:
I looked into code and I did not found any validation of `installroot` and `persistdir` option. The code somehow expects that the path is absolute in both cases. I suggest that at least `installroot` must be an absolute path. Then the check for installroot length is unneeded and installroot and persistdir can be always merged together.

Reply:
`installroot` must be an absolute path. And yes, I'll remove the `if`.